### PR TITLE
Move creating of invitations and grouping to the front

### LINKED ISSUE #170: Use a spinner when waiting for an invite link
I've seen this case a couple times:

screen shot 2015-09-02 at 4 16 56 pm

> Hang tight your students invitation is on the way!

It'd be better to wait or display a spinner so that the link is immediately available.

### PR DESCRIPTION
**Pull request by tarebyte**
Closes #170 

We shouldn’t let these objects be created in the background.

This makes the invitations be created as an ActiveRecord callback, and
the Grouping in a Transaction

Looking for implementation feedback

/cc @johndbritton

### CODE REVIEW

File: app/services/group_assignment_service.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    group_assignment = GroupAssignment.new(@new_group_assignment_params)
+
+    group_assignment.transaction do
+      if grouping.new_record?
+        grouping.save!
+        group_assignment.grouping = grouping
+      end
+
+      group_assignment.save!
+    end
```
Comment (johndbritton) [inline app/services/group_assignment_service.rb]: Doesn't rails automatically do a transaction if you do this:

```ruby
group_assignment.grouping = grouping
group_assignment.save!
```

File: app/services/group_assignment_service.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    @new_grouping_params         = new_grouping_params
+  end
+
+  # rubocop:disable MethodLength
+  def create_group_assignment
+    grouping = Grouping.find_by(id: @new_group_assignment_params[:grouping_id])
+    grouping = Grouping.new(@new_grouping_params) unless grouping.present?
+
+    group_assignment = GroupAssignment.new(@new_group_assignment_params)
+
```
Comment (johndbritton) [inline app/services/group_assignment_service.rb]: It'd be really cool if you could do something like this:

```ruby
group_assignment = GroupAssignment.new(@new_group_assignment_params)
group_assignment.grouping = Grouping.where(id: @new_group_assignment_params[:grouping_id]).first_or_create(@new_group_params)
group_assignment.save!
```

**Comment by johndbritton** [comment]
I think this is a good case for using the `build_*` methods. You can call `assignment.build_invitation` when you create the assignment and then both should be created when you persist the assignment.

File: app/services/group_assignment_service.rb (hunk 2 above)
Comment (tarebyte) [inline app/services/group_assignment_service.rb]: That would be pretty nice

File: app/controllers/assignments_controller.rb [hunk 3]
```diff
@@ -7,7 +7,6 @@ class AssignmentsController < ApplicationController
   decorates_assigned :organization
   decorates_assigned :assignment
 
-  rescue_from ActiveRecord::RecordInvalid, with: :error
```
Comment (tarebyte) [inline app/controllers/assignments_controller.rb]: Pro tip never override this

**Comment by tarebyte** [comment]
@johndbritton merging this for now, shout if you find any issues!

**Pull request merged**